pkg/crinit/standalone/standalone.go

```diff
@@ -0,0 +1,185 @@
+)
+
+type standaloneClusterRegistryOptions struct {
+	util.SubcommandOptions
+	apiServerServiceTypeString   string
+	apiServerEnableHTTPBasicAuth bool
+	apiServerEnableTokenAuth     bool
+}
+
+func (o *standaloneClusterRegistryOptions) Bind(flags *pflag.FlagSet) {
+	flags.StringVar(&o.apiServerServiceTypeString, util.APIServerServiceTypeFlag,
+		string(v1.ServiceTypeLoadBalancer),
+		"The type of service to create for the cluster registry. Options: 'LoadBalancer', 'NodePort'.")
+	flags.BoolVar(&o.apiServerEnableHTTPBasicAuth, "apiserver-enable-basic-auth", false,
+		"Enables HTTP Basic authentication for the cluster registry API server. Defaults to false.")
+	flags.BoolVar(&o.apiServerEnableTokenAuth, "apiserver-enable-token-auth", false,
+		"Enables token authentication for the cluster registry API server. Defaults to false.")
+}
+
+// NewCmdStandalone defines the `standalone` command that bootstraps a cluster registry
+// inside a host Kubernetes cluster.
+func NewCmdStandalone(cmdOut io.Writer, pathOptions *clientcmd.PathOptions, defaultServerImage, defaultEtcdImage string) *cobra.Command {
+	opts := &standaloneClusterRegistryOptions{}
+
+	cmd := &cobra.Command{
+		Use:     "standalone CLUSTER_REGISTRY_NAME --host-cluster-context=HOST_CONTEXT",
+		Short:   "Initialize a standalone cluster registry",
+		Long:    standalone_long,
+		Example: standalone_example,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := opts.SetName(args)
+			if err != nil {
+				glog.Fatalf("error: %v", err)
+			}
+
+			err = validateOptions(opts)
+			if err != nil {
+				glog.Fatalf("error: %v", err)
+			}
+
+			err = opts.MarshalOptions()
+			if err != nil {
+				glog.Fatalf("error: %v", err)
+			}
+
+			hostConfig, err := util.GetClientConfig(pathOptions, opts.Host, opts.Kubeconfig).ClientConfig()
+			if err != nil {
+				glog.Fatalf("error: %v", err)
+			}
+			hostClientset, err := client.NewForConfig(hostConfig)
+			if err != nil {
+				glog.Fatalf("error: %v", err)
+			}
+			err = Run(opts, cmdOut, hostClientset, pathOptions)
+			if err != nil {
+				glog.Fatalf("error: %v", err)
+			}
+		},
+	}
+
+	flags := cmd.Flags()
+	opts.BindCommon(flags, defaultServerImage, defaultEtcdImage)
+	opts.Bind(flags)
+
+	return cmd
+}
+
+type entityKeyPairs struct {
+	ca     *triple.KeyPair
+	server *triple.KeyPair
+	admin  *triple.KeyPair
+}
+
+type credentials struct {
+	username        string
+	password        string
+	token           string
+	certEntKeyPairs *entityKeyPairs
+}
+
+// validateOptions ensures that options are valid.
+func validateOptions(opts *standaloneClusterRegistryOptions) error {
+	opts.ApiServerServiceType = v1.ServiceType(opts.apiServerServiceTypeString)
+	return opts.ValidateCommonOptions()
+}
+
+// Run initializes a cluster registry.
+func Run(opts *standaloneClusterRegistryOptions, cmdOut io.Writer,
+	hostClientset client.Interface, pathOptions *clientcmd.PathOptions) error {
+
+	// Create Cluster Registry Namespace
```

Comment:
These inline comments might be TMI: the method names you've chosen are pretty clear.

Reply:
Removed.